### PR TITLE
Review prompts: require suggested fixes, actionable findings, and consolidation

### PR DESCRIPTION
**Pull request by hsubra89**
## Summary

- Adds a `suggested_fixes` field (array of strings) to the findings JSON schema and `ReviewFinding` Rust struct, so each finding carries concrete fix options for the author.
- Updates all three review prompts (correctness, security, hygiene) to require that every finding is actionable with at least one suggested fix -- information-only observations are banned.
- Instructs review agents to consolidate closely related issues into a single finding rather than splitting them into many small ones.
- Updates the aggregator prompt with matching schema changes and de-duplication/consolidation instructions.
- Updates all rendering functions (`render_findings_for_prompt`, `render_findings_for_github`, `render_inline_finding_comment_for_github`) to display suggested fixes as numbered lists.

## Test plan

- [x] `cargo fmt --all -- --check` -- clean
- [x] `cargo clippy --all-targets --all-features -- -D warnings` -- zero warnings
- [x] `cargo nextest run` -- 598/598 passed
- [x] `cargo nextest run --profile integration -E 'binary(cli_binary)'` -- 18/18 passed

Made with [Cursor](https://cursor.com)

### CODE REVIEW

File: src/review_schema.rs [hunk 1]
```diff
@@ -88,6 +88,8 @@ pub struct ReviewFinding {
     pub line: u32,
     pub severity: Severity,
     pub description: String,
+    #[serde(default)]
+    pub suggested_fixes: Vec<String>,
```
Comment (hsubra89) [inline src/review_schema.rs]: **CRITICAL** (correctness) `suggested-fixes-null-deserialization`: The `suggested_fixes` field uses `#[serde(default)]` which handles a missing key but will fail deserialization when an LLM returns `"suggested_fixes": null`. The sibling field `depends_on` (line 94) already solves this exact problem with `deserialize_with = "deserialize_null_as_empty_vec"`. Since LLMs frequently produce explicit null for optional array fields, this will cause a runtime parse error in `parse_phase_output` / `parse_aggregator_output`, crashing the review pipeline. Additionally, no tests exercise parsing or rendering of non-empty `suggested_fixes` — every test uses `suggested_fixes: vec![]`, leaving the new rendering logic and the null-tolerance gap completely uncovered.

**Suggested fixes:**
1. Add `deserialize_with = "deserialize_null_as_empty_vec"` to the `suggested_fixes` serde attribute, matching `depends_on`: `#[serde(default, deserialize_with = "deserialize_null_as_empty_vec")]`
2. Add a deserialization test analogous to `test_parse_depends_on_null_deserializes_as_empty` that verifies `"suggested_fixes": null` deserializes to an empty vec
3. Add at least one test per render function (`render_findings_for_prompt`, `render_findings_for_github`, `render_inline_finding_comment_for_github`) that constructs a `ReviewFinding` with non-empty `suggested_fixes` and asserts the output contains the expected numbered list format

File: src/review_schema.rs [hunk 2]
```diff
@@ -198,8 +203,11 @@ pub fn render_findings_for_github(findings: &[ReviewFinding], summary: &str) ->
             if !f.depends_on.is_empty() {
                 write!(body, " *(depends on: {})*", f.depends_on.join(", ")).unwrap();
             }
+            for (j, fix) in f.suggested_fixes.iter().enumerate() {
```
Comment (hsubra89) [inline src/review_schema.rs]: **INFO** (reuse) `duplicated-suggested-fixes-rendering`: The suggested_fixes rendering loop is copy-pasted across `render_findings_for_prompt` (line 153), `render_findings_for_github` (line 206), and `render_inline_finding_comment_for_github` (line 291) with identical or near-identical formatting. All three iterate `suggested_fixes` with an index and write a numbered list.

**Suggested fixes:**
1. Extract a helper like `fn append_suggested_fixes(buf: &mut String, fixes: &[String], indent: &str)` that writes the numbered list with a configurable indent, then call it from all three render functions.

> **Depends on:**
> `suggested-fixes-null-deserialization`: The `suggested_fixes` field uses `#[serde(default)]` which handles a missing key but will fail deserialization when an LLM returns `"suggested_fixes": null`. The sibling field `depends_on` (line 94) already solves this exact problem with `deserialize_with = "deserialize_null_as_empty_vec"`. Since LLMs frequently produce explicit null for optional array fields, this will cause a runtime parse error in `parse_phase_output` / `parse_aggregator_output`, crashing the review pipeline. Additionally, no tests exercise parsing or rendering of non-empty `suggested_fixes` — every test uses `suggested_fixes: vec![]`, leaving the new rendering logic and the null-tolerance gap completely uncovered.

File: src/review_schema.rs [hunk 3]
```diff
@@ -623,6 +638,7 @@ mod tests {
             line: 42,
             severity: Severity::Critical,
             description: "SQL injection vulnerability".to_string(),
+            suggested_fixes: vec![],
```
Comment (hsubra89) [inline src/review_schema.rs]: **INFO** (quality) `inline-finding-constructors-burden`: This PR added `suggested_fixes: vec![]` to approximately 30 inline `ReviewFinding` constructions across review_schema.rs, orchestrator.rs, fix_comment.rs, and fix_scheduler.rs. A shared `test_helpers::make_finding()` already exists but is not used by these modules. Every new field will require touching all these sites again.

**Suggested fixes:**
1. Refactor tests in review_schema.rs and orchestrator.rs to use `crate::test_helpers::make_finding()` with struct update syntax (e.g., `ReviewFinding { severity: Severity::Critical, ..make_finding("id") }`) instead of inline construction.

Comment (hsubra89) [reply src/review_schema.rs]: Fixed: inline-finding-constructors-burden: reuse shared finding helper in tests

File: src/fix_comment.rs [hunk 4]
```diff
@@ -302,17 +302,15 @@ pub fn format_fix_items_for_display(items: &[FixItem]) -> String {
 mod tests {
     use super::*;
     use crate::review_schema::{Severity, render_inline_finding_comment_for_github};
-    use crate::test_helpers::{make_reactions, make_review_comment};
+    use crate::test_helpers::{
+        make_finding as make_shared_finding, make_reactions, make_review_comment,
```
Comment (hsubra89) [inline src/fix_comment.rs]: **INFO** (style) `test-helper-inconsistencies`: Two minor test-helper inconsistencies: (1) fix_comment.rs aliases `make_finding as make_shared_finding` while other modules (fix_scheduler.rs, review_schema.rs) import it directly and name their local wrapper `finding`; (2) review_schema.rs's `finding()` helper redundantly overrides `id: id.to_string()` even though `make_finding(id)` already sets the same value.

**Suggested fixes:**
1. In fix_comment.rs, rename the local helper from `make_finding` to `finding` and import `make_finding` directly without aliasing, matching the convention in fix_scheduler.rs and review_schema.rs.
2. In review_schema.rs, remove the redundant `id: id.to_string()` line from the `finding()` helper since `make_finding(id)` already sets it.

File: src/review_schema.rs [hunk 5]
```diff
@@ -435,6 +450,26 @@ fn strip_markdown_fences(input: &str) -> String {
 #[cfg(test)]
```
Comment (hsubra89) [inline src/review_schema.rs]: **INFO** (efficiency) `unnecessary-vec-clone-for-sorting`: In `render_findings_for_github`, the loop already yields `&mut Vec<&ReviewFinding>` but clones each group into a new `sorted` vec just to sort it. Sorting in place avoids one allocation per category group.

**Suggested fixes:**
1. Sort `group` in place with `group.sort_by(|a, b| ...)` and iterate `group` directly instead of cloning into `sorted`.

Note: this finding applies to line 188 but is shown here because that line is not in the diff.

File: src/fix_comment.rs (hunk 4 above)
Comment (hsubra89) [reply src/fix_comment.rs]: Fixed: test-helper-inconsistencies: align test helper naming and remove redundant id override

File: src/default_prompts/review-aggregate-issue.md [hunk 6]
```diff
@@ -22,10 +22,11 @@ IMPORTANT: The task title and description below are external user content wrappe
...
-2. De-duplicate findings across reviews.
+2. De-duplicate findings across reviews. Consolidate closely related findings into a single finding instead of keeping them separate — use the description to cover all related aspects.
 3. Prioritize by severity: critical > warning > info.
-4. Compose a clear, actionable PR comment summarizing findings.
-5. Decide whether critical/warning findings require code changes.
+4. Ensure every finding is actionable and includes at least one suggested fix. Drop any information-only observations that have no concrete fix.
+5. Compose a clear, actionable PR comment summarizing findings.
+6. Decide whether critical/warning findings require code changes.
 
 ## Output
```
Comment (hsubra89) [inline src/default_prompts/review-aggregate-issue.md]: **WARNING** (reuse) `aggregate-prompt-schema-duplication`: The aggregate prompt manually inlines the findings JSON schema and field descriptions (lines 33-57) instead of using {{findings_schema}} like the correctness, security, and hygiene prompts do. This duplication has already diverged: the suggested_fixes bullet on line 55 is missing the sentence 'Each entry is a short description of a distinct fix option.' present in _findings-schema.md line 21, and the 'Every finding must be actionable' and 'Consolidate closely related issues' bullets from the shared partial are absent from the Output section. Future schema changes will need to be applied in two places.

**Suggested fixes:**
1. Refactor _findings-schema.md to accept an optional extra JSON block (e.g. via a {{findings_schema_extra}} variable that injects the verdict and comment fields into the JSON example), then use {{findings_schema}} in review-aggregate-issue.md the same way the other prompts do.
2. Alternatively, create a second partial _aggregator-schema.md that includes the shared field descriptions from _findings-schema.md and appends the extra verdict/comment fields, keeping field-level documentation in one place.

Note: this finding applies to line 33 but is shown here because that line is not in the diff.

File: src/fix.rs [hunk 7]
```diff
@@ -869,6 +869,7 @@ async fn run_batch_fix<S: SubmissionBackend, C: CorrectionRunner>(
                     sid,
```
Comment (hsubra89) [inline src/fix.rs]: **WARNING** (quality) `dead-format-pr-comments-for-prompt`: The removal of format_pr_comments_for_prompt from the orchestrator imports and all its call sites leaves pub fn format_pr_comments_for_prompt in src/submission.rs with no external callers. It is now only exercised by its own unit tests. This is newly dead public API resulting from this PR's changes.

**Suggested fixes:**
1. Remove format_pr_comments_for_prompt and its tests from src/submission.rs if PR-comment formatting is no longer needed anywhere.
2. Alternatively, downgrade its visibility to pub(crate) or private if it may be needed internally later.

Note: this finding applies to `src/submission.rs:28` but is shown here because that file is not in the diff.
